## Group Details

Group Name: ACG

Project Title: Best Suburb in Christchurch 

## Student Details

Student Name: Annabelle Bos

Class: DATA422 - Data Wrangling

Student ID: 61694371

## Step 1
Load the necessary libraries.

In [1]:
library(tidyverse)
library(dplyr)
library(skimr)

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.5     ✔ purrr   0.3.4
✔ tibble  3.1.4     ✔ dplyr   1.0.7
✔ tidyr   1.1.3     ✔ stringr 1.4.0
✔ readr   2.0.0     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



## Step 2
Obtain the directory of New Zealand schools by going to this website: https://www.educationcounts.govt.nz/directories/list-of-nz-schools#
Click the "Download the whole Directory" button. There does not seem to be any way to automate this function as the link behind the button is the same link as for the page itself. Attempting to use this to read the data directly results in the web page data being pulled through instead. Therefore, the user must download the directory first and ensure it is saved as directory.csv in the local folder to the programme.

The first 15 lines are skipped as they are information about the dataset rather then the data itself. There is also a blank row between the column headers and the data however, this filters out when the data is filtered to just the Christchurch schools.

In [2]:
schools_df <- read_csv(file = "directory.csv", skip = 15) 

schools_df

Rows: 2559 Columns: 42

── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (28): School Name, Telephone, Fax, Email^, Principal*, School Website, S...
dbl (14): School Number, Postal Code, Community of Learning ID, Latitude, Lo...


ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.



School Number School Name                        Telephone   Fax        
1    NA            NA                                 NA          NA         
2     1            Te Kura o Te Kao                   09 409 7813 NA         
3     2            Taipa Area School                  09 406 0159 09 406 1096
4     3            Kaitaia College                    09 408 0190 09 408 0193
5     4            Whangaroa College                  09 405 0199 09 405 0288
6     5            Kerikeri High School               09 407 8916 09 407 9323
7     6            Broadwood Area School              09 409 5878 09 409 5877
8     7            Okaihau College                    09 401 9030 09 401 9793
9     8            Bay of Islands College             09 404 1055 09 404 1048
10    9            Northland College                  09 401 3200 09 401 2378
11   10            Te Kura Taumata o Panguru          09 409 5701 09 409 5703
12   11            Opononi Area School                09 405 8500 09 405 8501
13   12            Tauraroa Area School               09 432 2643 09 432 2436
14   13            Kamo High School                   09 435 1688 09 435 0790
15   14            Tikipunga High School              09 437 3299 09 437 2522
16   15            Whangārei Boys' High School        09 430 4170 09 430 4172
17   16            Whangarei Girls' High School       09 430 4460 09 430 4410
18   17            Pompallier Catholic College        09 438 3950 09 430 0302
19   18            Mangakahia Area School             09 433 1702 NA         
20   19            Dargaville High School             09 439 7229 09 439 7563
21   20            Bream Bay College                  09 432 8226 09 432 8228
22   21            Otamatea High School               09 431 8230 09 431 8229
23   22            Ruawai College                     09 439 2216 09 439 2214
24   23            Rodney College                     09 423 6030 09 423 7555
25   24            Mahurangi College                  09 425 8039 09 425 7275
26   25            Orewa College                      09 427 3833 NA         
27   26            Kaipara College                    09 420 8640 09 420 7485
28   27            Long Bay College                   09 477 9009 09 477 9105
29   28            Rangitoto College                  09 477 0150 09 477 0077
30   29            Kristin School                     09 415 9566 NA         
⋮    ⋮             ⋮                                  ⋮           ⋮          
2530 6760          Somerville Intermediate School     09 535 1070 09 535 1071
2531 6763          Whangaparaoa College               09 424 9177 09 428 4794
2532 6783          Remarkables Primary School         03 441 4044 03 442 2052
2533 6788          Hobsonville Point Primary School   09 557 0810 NA         
2534 6920          Gulf Harbour School                09 428 0202 NA         
2535 6921          Point View School                  09 274 0637 NA         
2536 6922          Waiheke Primary School             09 372 2006 09 372 2376
2537 6929          Alfriston College                  09 269 0080 09 269 0083
2538 6930          Botany Downs Secondary College     09 273 2310 09 273 8551
2539 6932          Pinehill School (Browns Bay)       09 478 0301 09 478 0305
2540 6937          Summerland Primary                 09 836 7460 09 836 7461
2541 6939          Kohia Terrace School               09 630 4525 NA         
2542 6940          Te Akau ki Papamoa Primary School  07 542 2447 07 542 2448
2543 6941          Endeavour School                   07 855 5257 NA         
2544 6944          Randwick Park School               09 267 0112 09 267 0119
2545 6945          Selwyn Ridge School                07 544 1088 07 544 1081
2546 6946          Oteha Valley School                09 477 0033 09 476 7785
2547 6947          The Gardens School                 09 269 0041 NA         
2548 6948          Albany Junior High School          09 415 5473 NA         
2549 6955          Upper Ha

## Re-Naming Columns
As a group, it was decided that we should use a standard naming convention with underscores and lowercase letters. Therefore, the below is the renaming of all the columns to match this.

In [3]:
names(schools_df)[names(schools_df) == 'School Number' ] <- 'school_number'
names(schools_df)[names(schools_df) == 'School Name' ] <- 'school_name'
names(schools_df)[names(schools_df) == 'Telephone' ] <- 'telephone'
names(schools_df)[names(schools_df) == 'Fax' ] <- 'fax'
names(schools_df)[names(schools_df) == 'Email^'] <- 'email'
names(schools_df)[names(schools_df) == 'Principal*'] <- 'principal'
names(schools_df)[names(schools_df) == 'School Website'] <- 'school_website'
names(schools_df)[names(schools_df) == 'Street' ] <- 'street'
names(schools_df)[names(schools_df) == 'Suburb'] <- 'suburb'
names(schools_df)[names(schools_df) == 'Town / City'] <- 'town_city'
names(schools_df)[names(schools_df) == 'Postal Address'] <- 'postal_address'
names(schools_df)[names(schools_df) == 'Postal Address Suburb'] <- 'postal_suburb'
names(schools_df)[names(schools_df) == 'Postal Address City'] <- 'postal_city'
names(schools_df)[names(schools_df) == 'Postal Code'] <- 'postal_code'
names(schools_df)[names(schools_df) == 'Urban Area'] <- 'urban_area'
names(schools_df)[names(schools_df) == 'School Type'] <- 'school_type'
names(schools_df)[names(schools_df) == 'Definition' ] <- 'definition'
names(schools_df)[names(schools_df) == 'Authority' ] <- 'authority'
names(schools_df)[names(schools_df) == 'Donations' ] <- 'donations'
names(schools_df)[names(schools_df) == 'Gender of Students'] <- 'gender_of_students'
names(schools_df)[names(schools_df) == 'Territorial Authority'] <- 'territorial_authority'
names(schools_df)[names(schools_df) == 'Regional Council'] <- 'regional_council'
names(schools_df)[names(schools_df) == 'Ministry of Education Local Office'] <- 'moe_local_office'
names(schools_df)[names(schools_df) == 'Education Region'] <- 'education_region'
names(schools_df)[names(schools_df) == 'General Electorate'] <- 'general_electorate'
names(schools_df)[names(schools_df) == 'Māori Electorate'] <- 'māori_electorate'
names(schools_df)[names(schools_df) == 'Area Unit'] <- 'area_unit'
names(schools_df)[names(schools_df) == 'Ward'] <- 'ward'
names(schools_df)[names(schools_df) == 'Community of Learning ID'] <- 'comm_of_learning_id'
names(schools_df)[names(schools_df) == 'Community of Learning Name'] <- 'comm_of_learning_name'
names(schools_df)[names(schools_df) == 'Latitude'] <- 'latitude'
names(schools_df)[names(schools_df) == 'Longitude'] <- 'longitude'
names(schools_df)[names(schools_df) == 'Isolation Index'] <- 'isolation_index'
names(schools_df)[names(schools_df) == 'Decile'] <- 'decile'
names(schools_df)[names(schools_df) == 'Total School Roll'] <- 'total_school_roll'
names(schools_df)[names(schools_df) == 'European / Pākehā'] <- 'european_pākehā'
names(schools_df)[names(schools_df) == 'Māori'] <- "māori"
names(schools_df)[names(schools_df) == 'Pacific'] <- "pacific"
names(schools_df)[names(schools_df) == 'Asian'] <- "asian"
names(schools_df)[names(schools_df) == 'MELAA'] <- "m_e_l_a_a"
names(schools_df)[names(schools_df) == 'Other'] <- "other" 
names(schools_df)[names(schools_df) == 'International'] <- "international"

## Filtering for Christchurch Only
This dataset is very clean in its original form. This means that filtering by town_city for the Christchurch fields was a simple task. I checked against this list - https://geographic.org/streetview/new_zealand/canterbury_west_coast/christchurch/index.html - to see if it would be better to filter by suburbs, rather than by town_city. I found that there are duplicate suburbs between other cities, for example there is also a Northcote and an Avondale in Auckland. Therefore, I found it better to filter by city.

In [4]:
chch_schools <- schools_df %>%
  filter(town_city == 'Christchurch')

chch_schools

school_number school_name                                           
1    82           Aidanfield Christian School                           
2   315           St Bedes College                                      
3   316           Papanui High School                                   
4   317           Christchurch Adventist School                         
5   318           St Andrew's College (Christchurch)                    
6   319           Burnside High School                                  
7   320           Mairehau High School                                  
8   321           Shirley Boys' High School                             
9   324           Avonside Girls' High School                           
10  325           Rangi Ruru Girls' School                              
11  326           Villa Maria College                                   
12  327           Christchurch Boys' High School                        
13  328           Christchurch Girls' High School -Te Kura o Hine Waiora
14  330           Christ's College                                      
15  331           St Thomas of Canterbury College                       
16  333           St Margaret's College                                 
17  334           Riccarton High School                                 
18  335           Middleton Grange School                               
19  336           Hagley Community College                              
20  337           Linwood College                                       
21  338           Hornby High School                                    
22  339           Hillmorton High School                                
23  340           Cashmere High School                                  
24  341           Hillview Christian School                             
25  343           Marian College                                        
26  368           The Cathedral Grammar School                          
27  419           Rudolf Steiner School (Chch)                          
28  518           Kingslea School                                       
29  522           Halswell Residential College                          
30  531           Catholic Cathedral College                            
⋮   ⋮             ⋮                                                     
109 3508          South New Brighton School                             
110 3512          Spreydon School                                       
111 3517          St Albans Catholic School (Christchurch)              
112 3518          St Albans School                                      
113 3520          St Anne's School (Woolston)                           
114 3521          St Bernadette's School (Hornby)                       
115 3523          St James School (Aranui)                              
116 3531          St Joseph's School (Papanui)                          
117 3534          St Martins School                                     
118 3535          St Mary's School (Christchurch)                       
119 3537          St Patrick's School (Bryndwr)                         
120 3542          St Peter's School (Beckenham)                         
121 3543          St Teresa's School (Riccarton)                        
122 3544          Our Lady Star of the Sea School (Christchurch)        
123 3546          Sumner School                                         
124 3555          Templeton School                                      
125 3557          Thorrington School                                    
126 3571          Waimairi School                                       
127 3577          Wairakei School (Christchurch)                        
128 3581          Waltham School                                        
129 3585          Weedons School                                        
130 3588          Te Ara Koropiko West Spreydon School                  
131 3589          Westburn School                             

In [5]:
chch_schools %>% skim()

── Data Summary ────────────────────────
                           Values    
Name                       Piped data
Number of rows             138       
Number of columns          42        
_______________________              
Column type frequency:               
  character                28        
  numeric                  14        
________________________             
Group variables            None      

── Variable type: character ────────────────────────────────────────────────────
   skim_variable         n_missing complete_rate   min   max empty n_unique
 1 school_name                   0         1        11    54     0      138
 2 telephone                     0         1        11    11     0      136
 3 fax                          22         0.841    11    11     0      115
 4 email                         6         0.957    19    39     0      131
 5 principal                     0         1         8    33     0      136
 6 school_website                1       

## Missing Suburb Values
There are 138 schools in total directly in Christchurch. Of these, eight have no information in the suburb field. I have researched each of these instances individually and can fix this by pulling the Postal Address Suburb which on inspection is in the same vicinity for the blank suburbs.

In [6]:
chch_schools <- chch_schools %>% 
    mutate(suburb = coalesce(suburb,postal_suburb))

## Trimming White Space
There does not appear to be any whitespace throughout the dataset but to be certain, I have trimmed all strings of any possible whitespace.

In [7]:
chch_schools %>%
   mutate_if(is.character, str_trim)

school_number school_name                                           
1    82           Aidanfield Christian School                           
2   315           St Bedes College                                      
3   316           Papanui High School                                   
4   317           Christchurch Adventist School                         
5   318           St Andrew's College (Christchurch)                    
6   319           Burnside High School                                  
7   320           Mairehau High School                                  
8   321           Shirley Boys' High School                             
9   324           Avonside Girls' High School                           
10  325           Rangi Ruru Girls' School                              
11  326           Villa Maria College                                   
12  327           Christchurch Boys' High School                        
13  328           Christchurch Girls' High School -Te Kura o Hine Waiora
14  330           Christ's College                                      
15  331           St Thomas of Canterbury College                       
16  333           St Margaret's College                                 
17  334           Riccarton High School                                 
18  335           Middleton Grange School                               
19  336           Hagley Community College                              
20  337           Linwood College                                       
21  338           Hornby High School                                    
22  339           Hillmorton High School                                
23  340           Cashmere High School                                  
24  341           Hillview Christian School                             
25  343           Marian College                                        
26  368           The Cathedral Grammar School                          
27  419           Rudolf Steiner School (Chch)                          
28  518           Kingslea School                                       
29  522           Halswell Residential College                          
30  531           Catholic Cathedral College                            
⋮   ⋮             ⋮                                                     
109 3508          South New Brighton School                             
110 3512          Spreydon School                                       
111 3517          St Albans Catholic School (Christchurch)              
112 3518          St Albans School                                      
113 3520          St Anne's School (Woolston)                           
114 3521          St Bernadette's School (Hornby)                       
115 3523          St James School (Aranui)                              
116 3531          St Joseph's School (Papanui)                          
117 3534          St Martins School                                     
118 3535          St Mary's School (Christchurch)                       
119 3537          St Patrick's School (Bryndwr)                         
120 3542          St Peter's School (Beckenham)                         
121 3543          St Teresa's School (Riccarton)                        
122 3544          Our Lady Star of the Sea School (Christchurch)        
123 3546          Sumner School                                         
124 3555          Templeton School                                      
125 3557          Thorrington School                                    
126 3571          Waimairi School                                       
127 3577          Wairakei School (Christchurch)                        
128 3581          Waltham School                                        
129 3585          Weedons School                                        
130 3588          Te Ara Koropiko West Spreydon School                  
131 3589          Westburn School                             

## Areas of Interest
As a group we decided on the following areas of interest for each suburb:

Total school roll

Number of primary schools 

Number of intermediate schools

Number of secondary schools 

Average decile 

Flag for single sex schools 

Count of authorities (i.e. private / integrated / public) 

Flags were added for schools that met each of the criteria. I have opted to go very inclusive in this and include all schools that met the specifications, meaning if I was capturing whether or not there was a school available for intermediate aged children, I included solely intermediate schools, primary schools up to year 8 and secondary schools that started at year 7.

In [8]:
chch_schools <- chch_schools %>%
   mutate(primary_schools = ifelse((school_type == "Composite" | 
                                    school_type == "Contributing" | 
                                    school_type == "Full Primary"), 1, 0)) %>%
   mutate(intermediate_schools = ifelse((school_type == "Composite" | 
                                         school_type == "Full Primary" | 
                                         school_type == "Intermediate" |
                                         school_type == "Secondary (Year 7-15)"), 1, 0)) %>%
   mutate(secondary_schools = ifelse((school_type == "Composite" | 
                                         school_type == "Secondary (Year 9-15)" |
                                         school_type == "Secondary (Year 7-15)"), 1, 0)) %>%
   mutate(private_schools = ifelse(authority == "Private : Fully Registered", 1, 0)) %>%
   mutate(state_schools = ifelse((authority == "State" |
                                  authority == "State : Integrated"), 1, 0)) %>%
   mutate(co_educational = ifelse(gender_of_students == "Co-Educational", 1, 0)) %>%
   mutate(single_sex_girls = ifelse(gender_of_students == "Single Sex (Girls School)", 1, 0)) %>%
   mutate(single_sex_boys = ifelse(gender_of_students == "Single Sex (Boys School)", 1, 0)) 


chch_schools

school_number school_name                                           
1    82           Aidanfield Christian School                           
2   315           St Bedes College                                      
3   316           Papanui High School                                   
4   317           Christchurch Adventist School                         
5   318           St Andrew's College (Christchurch)                    
6   319           Burnside High School                                  
7   320           Mairehau High School                                  
8   321           Shirley Boys' High School                             
9   324           Avonside Girls' High School                           
10  325           Rangi Ruru Girls' School                              
11  326           Villa Maria College                                   
12  327           Christchurch Boys' High School                        
13  328           Christchurch Girls' High School -Te Kura o Hine Waiora
14  330           Christ's College                                      
15  331           St Thomas of Canterbury College                       
16  333           St Margaret's College                                 
17  334           Riccarton High School                                 
18  335           Middleton Grange School                               
19  336           Hagley Community College                              
20  337           Linwood College                                       
21  338           Hornby High School                                    
22  339           Hillmorton High School                                
23  340           Cashmere High School                                  
24  341           Hillview Christian School                             
25  343           Marian College                                        
26  368           The Cathedral Grammar School                          
27  419           Rudolf Steiner School (Chch)                          
28  518           Kingslea School                                       
29  522           Halswell Residential College                          
30  531           Catholic Cathedral College                            
⋮   ⋮             ⋮                                                     
109 3508          South New Brighton School                             
110 3512          Spreydon School                                       
111 3517          St Albans Catholic School (Christchurch)              
112 3518          St Albans School                                      
113 3520          St Anne's School (Woolston)                           
114 3521          St Bernadette's School (Hornby)                       
115 3523          St James School (Aranui)                              
116 3531          St Joseph's School (Papanui)                          
117 3534          St Martins School                                     
118 3535          St Mary's School (Christchurch)                       
119 3537          St Patrick's School (Bryndwr)                         
120 3542          St Peter's School (Beckenham)                         
121 3543          St Teresa's School (Riccarton)                        
122 3544          Our Lady Star of the Sea School (Christchurch)        
123 3546          Sumner School                                         
124 3555          Templeton School                                      
125 3557          Thorrington School                                    
126 3571          Waimairi School                                       
127 3577          Wairakei School (Christchurch)                        
128 3581          Waltham School                                        
129 3585          Weedons School                                        
130 3588          Te Ara Koropiko West Spreydon School                  
131 3589          Westburn School                             

## Missing Decile Values
Where there is no current decile information available for a school, the value has been recorded as 99. This means that any attempt to summarise by the mean of the decile will be thrown out by any schools with such a high value. To review these, I pulled them out of the dataset to inspect.

In [9]:
missing_value_schools <- subset(chch_schools, decile==99)
missing_value_schools

school_number school_name                   telephone   fax        
1  610          Seven Oaks School             03 377 8603 NA         
2  695          Seven Oaks Secondary School   03 377 8603 03 377 8603
3 2126          Jean Seabrook Memorial School 03 381 5383 03 381 5385
  email                       principal    school_website                 
1 office@sevenoaks.school.nz  Jeremy Orczy http://www.sevenoaks.school.nz/
2 office@sevenoaks.school.nz  Jeremy Orczy http://www.sevenoaks.school.nz 
3 school@seabrookmckenzie.net Mary Gillies NA                             
  street           suburb   town_city    ⋯ other international primary_schools
1 77 Murphys Road  Halswell Christchurch ⋯ 1     0             1              
2 77 Murphys Road  Halswell Christchurch ⋯ 0     0             0              
3 68 London Street Richmond Christchurch ⋯ 0     0             1              
  intermediate_schools secondary_schools private_schools state_schools
1 1                    0                 1               0            
2 0                    1                 1               0            
3 1                    0                 1               0            
  co_educational single_sex_girls single_sex_boys
1 1              0                0              
2 1              0                0              
3 1              0                0

In [10]:
missing_value_schools %>% skim()

── Data Summary ────────────────────────
                           Values    
Name                       Piped data
Number of rows             3         
Number of columns          50        
_______________________              
Column type frequency:               
  character                28        
  numeric                  22        
________________________             
Group variables            None      

── Variable type: character ────────────────────────────────────────────────────
   skim_variable         n_missing complete_rate   min   max empty n_unique
 1 school_name                   0         1        17    29     0        3
 2 telephone                     0         1        11    11     0        2
 3 fax                           1         0.667    11    11     0        2
 4 email                         0         1        26    27     0        2
 5 principal                     0         1        12    12     0        2
 6 school_website                1       

Given how few schools meet this criteria and their very low school rolls (maximum of 61 students at the largest school) I have opted to exclude these schools from the dataset completely.

In [11]:
chch_schools <- subset(chch_schools, decile!=99)
chch_schools

school_number school_name                                           
1    82           Aidanfield Christian School                           
2   315           St Bedes College                                      
3   316           Papanui High School                                   
4   317           Christchurch Adventist School                         
5   318           St Andrew's College (Christchurch)                    
6   319           Burnside High School                                  
7   320           Mairehau High School                                  
8   321           Shirley Boys' High School                             
9   324           Avonside Girls' High School                           
10  325           Rangi Ruru Girls' School                              
11  326           Villa Maria College                                   
12  327           Christchurch Boys' High School                        
13  328           Christchurch Girls' High School -Te Kura o Hine Waiora
14  330           Christ's College                                      
15  331           St Thomas of Canterbury College                       
16  333           St Margaret's College                                 
17  334           Riccarton High School                                 
18  335           Middleton Grange School                               
19  336           Hagley Community College                              
20  337           Linwood College                                       
21  338           Hornby High School                                    
22  339           Hillmorton High School                                
23  340           Cashmere High School                                  
24  341           Hillview Christian School                             
25  343           Marian College                                        
26  368           The Cathedral Grammar School                          
27  419           Rudolf Steiner School (Chch)                          
28  518           Kingslea School                                       
29  522           Halswell Residential College                          
30  531           Catholic Cathedral College                            
⋮   ⋮             ⋮                                                     
106 3508          South New Brighton School                             
107 3512          Spreydon School                                       
108 3517          St Albans Catholic School (Christchurch)              
109 3518          St Albans School                                      
110 3520          St Anne's School (Woolston)                           
111 3521          St Bernadette's School (Hornby)                       
112 3523          St James School (Aranui)                              
113 3531          St Joseph's School (Papanui)                          
114 3534          St Martins School                                     
115 3535          St Mary's School (Christchurch)                       
116 3537          St Patrick's School (Bryndwr)                         
117 3542          St Peter's School (Beckenham)                         
118 3543          St Teresa's School (Riccarton)                        
119 3544          Our Lady Star of the Sea School (Christchurch)        
120 3546          Sumner School                                         
121 3555          Templeton School                                      
122 3557          Thorrington School                                    
123 3571          Waimairi School                                       
124 3577          Wairakei School (Christchurch)                        
125 3581          Waltham School                                        
126 3585          Weedons School                                        
127 3588          Te Ara Koropiko West Spreydon School                  
128 3589          Westburn School                             

In [12]:
chch_schools %>% skim()

── Data Summary ────────────────────────
                           Values    
Name                       Piped data
Number of rows             135       
Number of columns          50        
_______________________              
Column type frequency:               
  character                28        
  numeric                  22        
________________________             
Group variables            None      

── Variable type: character ────────────────────────────────────────────────────
   skim_variable         n_missing complete_rate   min   max empty n_unique
 1 school_name                   0         1        11    54     0      135
 2 telephone                     0         1        11    11     0      134
 3 fax                          21         0.844    11    11     0      113
 4 email                         6         0.956    19    39     0      129
 5 principal                     0         1         8    33     0      134
 6 school_website                0       

## Writing the Tidied Dataset to CSV
We have decided to include the tidied dataset into the final product so have exported a copy to CSV prior to summarising. 

In [15]:
chch_schools %>%
  write_csv("invidual_schools_dataset.csv")

## Summarising the Dataset
Following the criteria determined by the group, I summarised the data by the given criterias and returned it in alphabetical order by suburb. I then wrote this to a CSV using the naming convention required for the next stage of the project where each dataset produced will be wrangled together.

In [ ]:
school_summary_by_suburb = chch_schools %>%
        select(suburb,
               total_school_roll,
               primary_schools,
               intermediate_schools,
               secondary_schools,
               decile,
               co_educational,
               single_sex_girls,
               single_sex_boys,
               private_schools,
               state_schools) %>%
        group_by(suburb) %>%
        summarise(number_of_schools = sum(single_sex_girls, single_sex_boys, co_educational),
                  total_school_roll = sum(total_school_roll), 
                  primary_schools = sum(primary_schools),
                  intermediate_schools = sum(intermediate_schools),
                  secondary_schools = sum(secondary_schools),
                  average_decile = mean(decile),
                  co_educational = sum(co_educational),
                  single_sex_girls = sum(single_sex_girls),
                  single_sex_boys = sum(single_sex_boys),
                  private_schools = sum(private_schools),
                  state_schools = sum(state_schools)) %>%
       arrange(suburb)

school_summary_by_suburb

In [ ]:
school_summary_by_suburb %>% skim()

In [ ]:
school_summary_by_suburb %>%
  write_csv("suburb_level_schools_summary.csv")

## Graphing
To add to the presentation, I wanted to provide an overall summary graph of the dataset. I have opted to show the total number of schools per suburb, ordered by the average school decile. 

This required setting the predetermined order of the data from alphabetically by suburb to be by the average decile field.

In [ ]:
school_summary_by_suburb$suburb <- factor(school_summary_by_suburb$suburb, levels = school_summary_by_suburb$suburb[order(school_summary_by_suburb$average_decile)])

In [ ]:
summary_graph <- school_summary_by_suburb %>%
  ggplot(aes(x = number_of_schools, y = suburb, fill = average_decile)) + 
    geom_bar(stat="identity", width=.5)  +
    labs(x = 'Count of Schools',y='Suburbs',
         title= "Number of Schools by Suburb and Average Decile")

summary_graph

In [ ]:
summary_graph %>%
  ggsave("school_summary_plt.jpg", ., device = NULL, scale = 1.0)